tests/kernel/update_kernel.pm

```diff
@@ -443,19 +433,18 @@ sub run {
     }
     elsif (get_var('AZURE')) {
         $kernel_package = 'kernel-azure';
-
-        if (get_var('AZURE_FIRST_RELEASE')) {
-            first_azure_release($repo);
-        }
-        else {
-            $self->prepare_azure;
-            update_kernel($repo, $incident_id);
-        }
+        $self->change_kernel_flavor($repo, $incident_id, $kernel_package,
+            ['kernel-default']);
     }
     elsif (get_var('KERNEL_BASE')) {
         $kernel_package = 'kernel-default-base';
-        $self->prepare_kernel_base;
-        update_kernel($repo, $incident_id);
+        $self->change_kernel_flavor($repo, $incident_id, $kernel_package,
+            ['kernel-default']);
+    }
+    elsif (get_var('KERNEL_64KB')) {
```

Comment:
I didn't like this approach before and I don't extension in same way. Why we should have one variable for each kernel flavor? It would be better to have just one variable which would contain kernel flavor.  Adding new flavor wouldn't need new variable, also line 434-447 duplicates a a lot.